torch/csrc/dynamo/eval_frame.c

```diff
@@ -745,17 +858,28 @@ static PyObject* _custom_eval_frame(
     return NULL;
   } else if (result != Py_None) {
     DEBUG_TRACE("create cache %s", get_frame_name(frame));
-    extra = create_cache_entry(extra, result);
+    CacheEntry* new_cache_entry = create_cache_entry(cache_entry, result);
     Py_DECREF(result);
-    set_cache_entry(frame->f_code, extra);
+    if (extra == NULL) {
+      // Create a new extra state as this is the first time we see this code object.
+      create_and_set_extra_state(frame->f_code, new_cache_entry, frame_state);
+    } else {
+      // Update the existing cache_entry on the extra scratch space.
+      override_cache_entry_on_extra(frame->f_code, extra, new_cache_entry);
```

Comment:
This seems extremely suspect. override_cache_entry_on_extra sets cache entry. But extra is already set to your malloc field.

The problem is that when you consolidated the extra and frame state fields, you cargo culted the setter logic. But you do not need to set the NEW extra entry anymore, because modifying the cache entry field on the struct is sufficient!

Reply:
Yes, you are right. I realized this well that I am doing something very dumb.